clients/drcachesim/drpt2trace/pt2ir.h

```diff
@@ -0,0 +1,316 @@
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* pt2ir: convert a PT raw trace to DynamoRIO's IR format. */
+
+#ifndef _PT2IR_H_
+#define _PT2IR_H_ 1
+
+/**
+ * @file drmemtrace/pt2ir.h
+ * @brief Offline PT raw trace converter.
+ */
+
+#include <string>
+#include <vector>
+#include <memory>
+#define DR_FAST_IR 1
+#include "dr_api.h"
+
+#ifndef IN
+#    define IN // nothing
+#endif
+#ifndef OUT
+#    define OUT // nothing
+#endif
+#ifndef INOUT
+#    define INOUT // nothing
+#endif
+
+/**
+ * The type of pt2ir_t::convert() return value.
+ */
+enum pt2ir_convert_status_t {
+    PT2IR_CONV_SUCCESS = 0,       /**< The conversion process is successful. */
+    PT2IR_CONV_ERROR_SYNC_PACKET, /**< The conversion process ends with a failure to sync
+                                   *   to the PSB packet.
+                                   */
+    PT2IR_CONV_ERROR_HANDLE_SIDEBAND_EVENT, /**< The conversion process ends with a
+                                             *   failure to handle a perf event.
+                                             */
+    PT2IR_CONV_ERROR_GET_PENDING_EVENT, /**< The conversion process ends with a failure to
+                                         *   get the pending event.
+                                         */
+    PT2IR_CONV_ERROR_SET_IMAGE, /**< The conversion process ends with a failure to set the
+                                 *   new image.
+                                 */
+    PT2IR_CONV_ERROR_DECODE_NEXT_INSTR /**< The conversion process ends with a failure to
+                                        *   decode the next intruction.
+                                        */
+};
+
+/**
+ * The types of the CPU vendor.
+ */
+enum pt_cpu_vendor_t {
+    /**
+     * The CPU vendor is unknown.
```

Comment:
Note that these can be a single line if you like: a little more concise.
```
/** The CPU vendor is unknown. */
```

Reply:
Done.